rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/core/RMCore.java

```diff
@@ -1341,70 +1250,103 @@ public BooleanWrapper createNodeSource(String nodeSourceName, String infrastruct
+    protected void prepareNodeSource(NodeSourceData persistedNodeSource) {
+        InfrastructureManager infrastructureManager = InfrastructureManagerFactory.create(persistedNodeSource.toNodeSourceDescriptor());
 
-        //checking that nsname doesn't contain invalid characters and doesn't exist yet
-        checkNodeSourceName(nodeSourceName);
+        NodeSourcePolicy notActivePolicy = NodeSourcePolicyFactory.create(persistedNodeSource.getPolicyType());
 
-        logger.info("Creating a node source : " + nodeSourceName);
+        NodeSource nodeSource = new NodeSource(this.getUrl(),
+                                               persistedNodeSource.getName(),
+                                               infrastructureManager,
+                                               notActivePolicy,
+                                               (RMCore) PAActiveObject.getStubOnThis(),
+                                               this.monitoring,
+                                               persistedNodeSource.toNodeSourceDescriptor());
 
-        boolean existPersistedNodes = existPersistedNodes(nodeSourceName, nodesRecoverable);
+        String nodeSourceName = nodeSource.getName();
 
-        InfrastructureManager im;
+        this.undeployedNodeSources.put(nodeSourceName, nodeSource);
 
-        // we need to reload the infrastructure variables saved in database if
-        // we recover the nodes
-        if (existPersistedNodes) {
-            im = InfrastructureManagerFactory.recover(nodeSourceData);
-            if (!im.getDeployingAndLostNodes().isEmpty()) {
-                dbManager.removeAllNodesFromNodeSource(nodeSourceName);
-                im = InfrastructureManagerFactory.create(nodeSourceData);
-                existPersistedNodes = false;
-            }
-        } else {
-            im = InfrastructureManagerFactory.create(nodeSourceData);
+        this.monitoring.nodeSourceEvent(new RMNodeSourceEvent(RMEventType.NODESOURCE_DEFINED,
+                                                              persistedNodeSource.getProvider().getName(),
+                                                              nodeSourceName,
+                                                              nodeSource.getDescription(),
+                                                              persistedNodeSource.getProvider().getName(),
+                                                              nodeSource.getDescriptor().getStatus().toString()));
+
+        logger.info("The node source " + nodeSourceName + " has been successfully defined by " +
+                    persistedNodeSource.getProvider().getName());
+    }
+
+    /**
+     * @deprecated  As of version 8.1, replaced by {@link #defineNodeSource(String, String, Object[], String, Object[],
+     * boolean)} and {@link #deployNodeSource(String)}
+     */
+    @Deprecated
+    public BooleanWrapper createNodeSource(String nodeSourceName, String infrastructureType,
```

Comment:
![INFO](https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO') Do not forget to remove this deprecated code someday. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1133)